mlc_llm/relax_model/gpt_neox.py

```diff
@@ -32,89 +32,33 @@
 
 @dataclass
 class GPTNeoXConfig:  # pylint: disable=too-many-instance-attributes
-    use_parallel_residual: bool
-    hidden_size: int
-    intermediate_size: int
-    num_attention_heads: int
-    num_hidden_layers: int
-    vocab_size: int
-    rotary_pct: float = 0.25
-    rotary_emb_base: int = 10000
-
-    dtype: str = "float32"
-    layer_norm_eps: float = 1e-05
-    max_sequence_length: int = 2048
-
-
-MODEL_CONFIG = {
-    "dolly-v2-3b": {
-        "use_parallel_residual": True,
-        "hidden_size": 2560,
-        "intermediate_size": 10240,
-        "num_attention_heads": 32,
-        "num_hidden_layers": 32,
-        "vocab_size": 50280,
-    },
-    "dolly-v2-7b": {
-        "use_parallel_residual": True,
-        "hidden_size": 4096,
-        "intermediate_size": 16384,
-        "num_attention_heads": 32,
-        "num_hidden_layers": 32,
-        "vocab_size": 50280,
-    },
-    "dolly-v2-12b": {
-        "use_parallel_residual": True,
-        "hidden_size": 5120,
-        "intermediate_size": 20480,
-        "num_attention_heads": 40,
-        "num_hidden_layers": 36,
-        "vocab_size": 50280,
-    },
-    "stablelm-tuned-alpha-3b": {
-        "use_parallel_residual": True,
-        "hidden_size": 4096,
-        "intermediate_size": 16384,
-        "num_attention_heads": 32,
-        "num_hidden_layers": 16,
-        "vocab_size": 50688,
-    },
-    "stablelm-tuned-alpha-7b": {
-        "use_parallel_residual": True,
-        "hidden_size": 6144,
-        "intermediate_size": 24576,
-        "num_attention_heads": 48,
-        "num_hidden_layers": 16,
-        "vocab_size": 50432,
-    },
-    "RedPajama-INCITE-Base-3B-v1": {
-        "use_parallel_residual": False,
-        "hidden_size": 2560,
-        "intermediate_size": 10240,
-        "num_attention_heads": 32,
-        "num_hidden_layers": 32,
-        "vocab_size": 50432,
-        "rotary_pct": 1.0,
-    },
-    "RedPajama-INCITE-Chat-3B-v1": {
-        "use_parallel_residual": False,
-        "hidden_size": 2560,
-        "intermediate_size": 10240,
-        "num_attention_heads": 32,
-        "num_hidden_layers": 32,
-        "vocab_size": 50432,
-        "rotary_pct": 1.0,
-    },
-    "RedPajama-INCITE-Instruct-3B-v1": {
-        "use_parallel_residual": False,
-        "hidden_size": 2560,
-        "intermediate_size": 10240,
-        "num_attention_heads": 32,
-        "num_hidden_layers": 32,
-        "vocab_size": 50432,
-        "rotary_pct": 1.0,
-    },
-}
+    def __init__(
```

Comment:
Hi Sudeep, these config classes are decorated by `dataclass` and therefore have implicit constructor. This means we just need to list the contents of the class and don’t need the `__init__` explicitly. You can refer to the documentation of dataclass https://docs.python.org/3/library/dataclasses.html.

Reply:
@MasterJH5574 Thanks, I changed it to the explicit init because we needed the `**kwargs` to capture the additional arguments from the config json. It looks like supporting kwargs with the dataclass seems less clean than just having an explicit init (see [here](https://stackoverflow.com/questions/55099243/python3-dataclass-with-kwargsasterisk)). Let me know what you think.

Reply:
I see. Then let’s go with yours.